data_bounds.py

```diff
@@ -0,0 +1,26 @@
+# List of valid rate selections from high to low in Hz
+VALID_RATES = [120, 10, 1, 0]
+
+def get_valid_rate(rate):
+    """
+    Return the real rate that we'll request from ACR.
+
+    Not all rate requests are valid, there is a strict
+    set that we are allowed to request. Requesting an
+    inbetween value is the same as requesting the next
+    lowest available value.
+
+    Parameters
+    ----------
+    rate : int
+        The rate in Hz that we want to request.
+
+    Returns
+    -------
+    valid_rate : int
+        The rate in Hz that is actually requested.
+    """
+    for valid_rate in VALID_RATES:
+        if rate >= valid_rate:
+            return valid_rate
+    return 0
```

Comment:
A negative rate seems like a weird upstream bug waiting to happen, but perhaps here isn't where it should be caught 🤷‍♂️

Reply:
I guess I didn't think too hard about it, though I clearly knew it could happen when I wrote this function. I don't think I ever explicitly tested what putting a negative rate into the line_beam_parameters would do, nor do I have a way of testing it right now. I'll double check what PMPS lib does with a negative rate value before proceeding.

Reply:
I think this will end up being OK, negative rate requests are transmitted as zero: https://github.com/pcdshub/lcls-twincat-pmps/blob/cc64ac08ec4454a335f342e653fc0910669af992/lcls-twincat-pmps/PMPS/MajorComponents/PMPS_MPS_Interface/FB_BeamClassOutputs.TcPOU#L49-L58